File: submission-tools/create_submission.py
"""Create a 12-row CSV submission template using only the Python standard library."""

from __future__ import annotations

import argparse
import csv
from datetime import date
from pathlib import Path

COLUMNS = (
    "reference_date",
    "target",
    "horizon",
    "location",
    "output_type",
    "output_type_id",
    "value",
)


def main() -> None:
    parser = argparse.ArgumentParser(description=__doc__)
    parser.add_argument("reference_date")
    parser.add_argument("model_id")
    parser.add_argument("--output", type=Path)
    args = parser.parse_args()
    reference_date = date.fromisoformat(args.reference_date)
    if reference_date.weekday() != 6:
        parser.error("reference_date must be a Sunday")
    output = args.output or (
        Path("model-output")
        / args.model_id
        / f"{reference_date.isoformat()}-{args.model_id}.csv"
    )
    output.parent.mkdir(parents=True, exist_ok=True)
    with output.open("w", encoding="utf-8", newline="") as handle:
        writer = csv.DictWriter(handle, fieldnames=COLUMNS)
        writer.writeheader()
        for location in ("SE", "SE-M", "SE-O"):
            for horizon in range(4):
                writer.writerow(
                    {
                        "reference_date": reference_date.isoformat(),
                        "target": "weekly incident influenza cases",
                        "horizon": horizon,
                        "location": location,
                        "output_type": "mean",
                        "output_type_id": "",
                        "value": "",
                    }
                )
    print(output)


if __name__ == "__main__":
    main()

File: submission-tools/create_submission.R
args <- commandArgs(trailingOnly = TRUE)
if (length(args) < 2) {
  stop("Usage: Rscript submission-tools/create_submission.R YYYY-MM-DD team-model [output.csv]")
}

reference_date <- as.Date(args[[1]])
if (is.na(reference_date) || as.POSIXlt(reference_date)$wday != 0) {
  stop("reference_date must be a Sunday in YYYY-MM-DD format")
}
model_id <- args[[2]]
output <- if (length(args) >= 3) args[[3]] else file.path(
  "model-output", model_id, paste0(reference_date, "-", model_id, ".csv")
)
dir.create(dirname(output), recursive = TRUE, showWarnings = FALSE)

submission <- expand.grid(
  horizon = 0:3,
  location = c("SE", "SE-M", "SE-O"),
  KEEP.OUT.ATTRS = FALSE,
  stringsAsFactors = FALSE
)
submission$reference_date <- as.character(reference_date)
submission$target <- "weekly incident influenza cases"
submission$output_type <- "mean"
submission$output_type_id <- NA_character_
submission$value <- NA_real_
submission <- submission[c(
  "reference_date", "target", "horizon", "location",
  "output_type", "output_type_id", "value"
)]
write.csv(submission, output, row.names = FALSE, na = "")
message(output)
